gateway/api/v1/route_registry.py

```diff
@@ -0,0 +1,34 @@
+"""
+RouteRegistry Module
+"""
+
+from typing import Callable, List, Optional
+from django.urls import path
+from django.urls.resolvers import URLPattern
+
+
+class RouteRegistry:
+    """
+    RouteRegistry class is used for register routes like views endpoints.
+    """
+
+    _routes: List[URLPattern] = []
+
+    @classmethod
+    def register(cls, url_path: str, view_func: Callable, name: Optional[str] = None):
+        """
+        register a route to serve as endpoint.
+        """
+        route_name = name or view_func.__name__.replace("_", "-")
```

Comment:
Is this replacement really needed @korgan00 ? Just curious in case we need to add a regex as a parameter in the URL that requires to use `_`.